lib/sequel/model/exceptions.rb

```diff
@@ -24,11 +24,18 @@ def initialize(message=nil, model=nil)
   UndefinedAssociation = Class.new(Error)
   ).name
 
-  (
   # Raised when a mass assignment method is called in strict mode with either a restricted column
   # or a column without a setter method.
-  MassAssignmentRestriction = Class.new(Error)
-  ).name
+  class MassAssignmentRestriction < Error
+    # The Sequel::Model object related to this exception.
+    attr_reader :model
+
+    def self.create(msg, model)
+      @model = model
```

Comment:
This sets the instance variable on the class, not on the created instance.  You don't catch this because you only test for the message and not for the model value.  The tests should be updated to test that the model is set correctly.